days/src/day8/Day8.kt

```diff
@@ -0,0 +1,73 @@
+package day8
+
+import Day
+import java.io.File
+import java.util.Collections.copy
+import java.util.concurrent.atomic.DoubleAccumulator
+
+class Day8(private val program: List<Instruction>) : Day {
+    enum class OPERATION {
+        ACC, JMP, NOP;
+
+        override fun toString(): String {
+            return super.toString().toLowerCase()
+        }
+    }
+    fun OPERATION.execute(state: State,amount: Int) = when(this) {
+        OPERATION.ACC -> State(state.accumulator + amount, state.pc + 1)
+        OPERATION.JMP -> State(state.accumulator, state.pc + amount)
+        OPERATION.NOP -> State(state.accumulator, state.pc + 1)
+    }
+    data class State(val accumulator: Int = 0, val pc: Int = 0)
```

Comment:
Passing the state along the execution path as you go is interesting. I did a more traditional OOP version and wrote a function that takes in a program and does the full execution (and internally has the pc and accumulator which it returns). From what I remember last year though that became kinda an issue because it didn't give a lot of program execution introspection which this doesn't suffer from.

Reply:
I did that originally, I felt slight code smell from having my `OP.execute` extension function messing with internal state of the outer object. No real smell there, but something felt off so I tried this instead, doing state transformation.